src/function/table/arrow_conversion.cpp

```diff
@@ -120,55 +120,124 @@ static void ColumnArrowToDuckDBDictionary(Vector &vector, ArrowArray &array, Arr
                                           idx_t size, const ArrowType &arrow_type, int64_t nested_offset = -1,
                                           ValidityMask *parent_mask = nullptr, uint64_t parent_offset = 0);
 
-static void ArrowToDuckDBList(Vector &vector, ArrowArray &array, ArrowArrayScanState &array_state, idx_t size,
-                              const ArrowType &arrow_type, int64_t nested_offset, ValidityMask *parent_mask,
-                              int64_t parent_offset) {
-	auto size_type = arrow_type.GetSizeType();
-	idx_t list_size = 0;
-	auto &scan_state = array_state.state;
+namespace {
 
-	SetValidityMask(vector, array, scan_state, size, parent_offset, nested_offset);
+struct ArrowListOffsetData {
+	idx_t list_size = 0;
 	idx_t start_offset = 0;
+};
+
+} // namespace
+
+template <class BUFFER_TYPE>
+static ArrowListOffsetData ConvertArrowListOffsetsTemplated(Vector &vector, ArrowArray &array, idx_t size,
+                                                            idx_t effective_offset) {
+	ArrowListOffsetData result;
+	auto &start_offset = result.start_offset;
+	auto &list_size = result.list_size;
+
 	idx_t cur_offset = 0;
+	auto offsets = ArrowBufferData<BUFFER_TYPE>(array, 1) + effective_offset;
+	start_offset = offsets[0];
+	auto list_data = FlatVector::GetData<list_entry_t>(vector);
+	for (idx_t i = 0; i < size; i++) {
+		auto &le = list_data[i];
+		le.offset = cur_offset;
+		le.length = offsets[i + 1] - offsets[i];
+		cur_offset += le.length;
+	}
+	list_size = offsets[size];
+	list_size -= start_offset;
+	return result;
+}
+
+template <class BUFFER_TYPE>
+static ArrowListOffsetData ConvertArrowListViewOffsetsTemplated(Vector &vector, ArrowArray &array, idx_t size,
+                                                                idx_t effective_offset) {
+	ArrowListOffsetData result;
+	auto &start_offset = result.start_offset;
+	auto &list_size = result.list_size;
+
+	list_size = 0;
+	auto offsets = ArrowBufferData<BUFFER_TYPE>(array, 1) + effective_offset;
+	auto sizes = ArrowBufferData<BUFFER_TYPE>(array, 2) + effective_offset;
+
+	// In ListArrays the offsets have to be sequential
+	// ListViewArrays do not have this same constraint
+	// for that reason we need to keep track of the lowest offset, so we can skip all the data that comes before it
+	// when we scan the child data
+	auto lowest_offset = offsets[0];
```

Comment:
Is this safe if the length of the array is `0`? list-views don't have `length+1` offsets like regular lists.

Reply:
You raise a good point, I think this does indeed blow up on empty lists because of this, I'll fix it

Reply:
I couldn't test this because we don't hit this path if we're scanning a zero-length array, pyarrow can't seem to produce nested lists where the inner list is `[]`, something about failing to infer type.

When we produce an empty nested list we initialize the allocation higher depending on the size of the parent list, so I also couldn't get it to break there.

None the less, this looks like an out of bounds access so I've prevented it with a ternary 👍 
Thanks for spotting it